### PR TITLE
AP_FOLLOW: Initialize offsets based on offset type

### PR DESCRIPTION
**Pull request by hoangthien94**
This PR fixes the following:
- Definition for `AP_FOLLOW_OFFSET_TYPE_RELATIVE`: from `0` to `1`.
- Initialize offsets properly when `FOLL_OFS_TYPE = 1` (NED = 0, relative = 1), a fix for this issue https://github.com/ArduPilot/ardupilot/issues/8915.

### CODE REVIEW

File: libraries/AP_Follow/AP_Follow.cpp [hunk 1]
```diff
@@ -331,10 +331,25 @@ bool AP_Follow::get_velocity_ned(Vector3f &vel_ned, float dt) const
 
 // initialise offsets to provided distance vector to other vehicle (in meters in NED frame) if required
 void AP_Follow::init_offsets_if_required(const Vector3f &dist_vec_ned)
-{
+{
```
Comment (amilcarlucas) [inline libraries/AP_Follow/AP_Follow.cpp]: Whitespace

**Comment by rmackay9** [comment]
Ah, very nice to see this.  Hopefully I'll review completely tomorrow.  txs!

**Comment by rmackay9** [comment]
@hoangthien94, you've tested this in SITL?

Just a hunch but it seems to me that the rotation angle should be reversed in the two situations.  I.e. in one direction while saving, in the opposite direction when applying..

**Comment by rmackay9** [comment]
I'm closing this PR because I've created a [new one](https://github.com/ArduPilot/ardupilot/pull/9019) that includes this PR but resolves a couple of things I found.  @hoangthien94, if you could peak at the new PR and confirm you think it's correct that would be a great help, thanks!

**Pull request closed without merging**